### PR TITLE
Add hashing support to BlockHeader

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Needed for https://github.com/bpfkorea/agora/pull/67

### CODE REVIEW

File: source/agora/common/Block.d [hunk 1]
```diff
@@ -54,4 +94,44 @@ public struct Block
...
+    /***************************************************************************
+
+        Implements hashing support
+
+        Params:
+            dg = hashing function
+
+    ***************************************************************************/
+
+    public void computeHash (scope HashDg dg) const nothrow @safe @nogc
```
Comment (Geod24) [inline source/agora/common/Block.d]: Why do you need this ?

Comment (AndrejMitrovic) [reply source/agora/common/Block.d]: Right, I only need the hash of the previous header, which contains the transaction hash (merkle root later). Let me remove this.

**Comment by AndrejMitrovic** [comment]
Removed block hashing.

**Review by Geod24 — APPROVED** [review]

File: source/agora/common/Block.d [hunk 2]
```diff
@@ -37,6 +38,45 @@ public struct BlockHeader
...
+{
+    import agora.common.crypto.Key;
+    auto address = `GDD5RFGBIUAFCOXQA246BOUPHCK7ZL2NSHDU7DVAPNPTJJKVPJMNLQFW`;
+    PublicKey pubkey = PublicKey.fromString(address);
+
+    // above parts not @safe/@nogc yet
+    () @safe @nogc nothrow
+    {
+        Output[1] outputs = [Output(100, pubkey)];
+        Transaction tx = { outputs: outputs[]};
```
Comment (Geod24) [inline source/agora/common/Block.d]:
```suggestion
        Transaction tx = { outputs: outputs[] };
```

File: source/agora/common/Block.d [hunk 3]
```diff
@@ -37,6 +38,45 @@ public struct BlockHeader
...
+unittest
+{
+    import agora.common.crypto.Key;
+    auto address = `GDD5RFGBIUAFCOXQA246BOUPHCK7ZL2NSHDU7DVAPNPTJJKVPJMNLQFW`;
+    PublicKey pubkey = PublicKey.fromString(address);
+
+    // above parts not @safe/@nogc yet
+    () @safe @nogc nothrow
+    {
+        Output[1] outputs = [Output(100, pubkey)];
```
Comment (Geod24) [inline source/agora/common/Block.d]:
```suggestion
        Output[1] outputs = [ Output(100, pubkey) ];
```

If you want to keep it consistent? The style guide does not cover this.

File: source/agora/common/Block.d [hunk 4]
```diff
@@ -37,6 +38,45 @@ public struct BlockHeader
...
     /// (later to be replaced with a merkle root)
     public Hash tx_hash;
+
+
+    /***************************************************************************
+
+        Implements hashing support
+
+        Params:
+            dg = hashing function
```
Comment (Geod24) [inline source/agora/common/Block.d]:
```suggestion
            dg = Hashing function accumulator
```

File: source/agora/common/Block.d (hunk 3 above)
Comment (AndrejMitrovic) [reply source/agora/common/Block.d]: I'm Switzerland (neutral) on this one.

**Pull request merged**